$OpenBSD: patch-src_floating_c,v 1.14 2012/10/01 08:12:17 dcoppa Exp $

From 8480b4cb58c30a5fa4ff9761aaed198efa8429f8 Mon Sep 17 00:00:00 2001
From: Michael Stapelberg <michael@stapelberg.de>
Date: Fri, 28 Sep 2012 16:23:01 +0000
Subject: Bugfix: don't crash when dragged floating window closes

Previously, while the loop would not be executed anymore, the
callback (for actually repositioning the window after a MotionNotify)
would still get called once. This commit avoids that, thus fixing
the crash.

--- src/floating.c.orig	Wed Sep 19 18:08:09 2012
+++ src/floating.c	Mon Oct  1 09:42:59 2012
@@ -570,7 +570,7 @@ void drag_pointer(Con *con, const xcb_button_press_eve
                 free(inside_event);
         } while ((inside_event = xcb_poll_for_event(conn)) != NULL);
 
-        if (last_motion_notify == NULL)
+        if (last_motion_notify == NULL || loop_done)
             continue;
 
         new_x = ((xcb_motion_notify_event_t*)last_motion_notify)->root_x;
